torch/csrc/autograd/variable.cpp

```diff
@@ -154,6 +154,25 @@ AutogradMeta* materialize_autograd_meta(const at::TensorBase& self) {
   return get_autograd_meta(self);
 }
 
+void update_retains_grad(
```

Comment:
This is actually updating both hooks and retains_grad right?

Reply:
Renamed to "update_cpp_hooks_on_new_gradfn"